kernel/src/scan/mod.rs

```diff
@@ -716,6 +719,63 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_replay_for_scan_data() {
+        let path = std::fs::canonicalize(PathBuf::from("./tests/data/parquet_row_group_skipping/"));
+        let url = url::Url::from_directory_path(path.unwrap()).unwrap();
+        let engine = SyncEngine::new();
+
+        let table = Table::new(url);
+        let snapshot = table.snapshot(&engine, None).unwrap();
+        let scan = snapshot.into_scan_builder().build().unwrap();
+        let data: Vec<_> = scan
+            .replay_for_scan_data(&engine)
+            .unwrap()
+            .try_collect()
+            .unwrap();
+        // No predicate pushdown attempted, because at most one part of a multi-part checkpoint
+        // could be skipped when looking for adds/removes.
+        assert_eq!(data.len(), 5);
+    }
+
+    #[test]
+    fn test_data_row_group_skipping() {
```

Comment:
These two tests are newly added today, and verify that row group skipping actually occurred. 

NOTE: These new tests all rely on the parquet reader to return one `EngineData` per row group, but that should always be true since the parquet files being read have fewer than ten rows (and the checkpoint part files have just one row).

Reply:
maybe add note above as comment in the test for posterity?